File: Advanced Methods of Programming/Laborator1JAVA/src/main/java/domain/Proba.java
package domain;

public class Proba implements HasId<Integer> {
    private int idProba;
    private Distanta distanta;
    private Stil stil;

    public Proba(Integer idProba, Integer distanta, String stil){
        this.idProba=idProba;
        this.stil=Stil.valueOf(stil);
        this.distanta=Distanta.valueOf(Distanta.getValue(distanta));
    }

    @Override
    public Integer getId() {
        return idProba;
    }

    @Override
    public void setId(Integer integer) { this.idProba=integer; }

    public Distanta getDistanta() {
        return distanta;
    }

    public void setDistanta(Distanta distanta) {
        this.distanta = distanta;
    }

    public Stil getStil() {
        return stil;
    }

    public void setStil(Stil stil) {
        this.stil = stil;
    }

    public Integer getDistantaInt(){
        return Distanta.getValueInt(this.distanta.toString());
    }
}

File: Advanced Methods of Programming/Lab1C-MPP/Lab1C-MPP/Domain/Proba.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab1C_MPP.Domain
{
    class Proba : IHasID<int>
    {
        private int IdProba;
        private string Stil;
        private int Distanta;
       public Proba(int Id, string St, int Dist)
        {
            this.IdProba = Id;
            this.Stil = St;
            this.Distanta = Dist;
        }

        public int GetId() { return this.IdProba; }

        public string GetStil() { return this.Stil; }

        public int GetDistanta() { return this.Distanta; }
    }
}
